### ISSUE TITLE
Rustc runs out of memory when dealing with extremely large array literals in macros

### ISSUE THREAD

**Issue by camlorn** [rust-lang/rust#35154]
Playground link: https://is.gd/WsYvhb

This was originally a typo having to do with using a screen reader, which doesn't read exclamations in positions which it decides aren't the ends of sentences well with my currently configured settings.  I first observed it locally with 0.10 i686-pc-windows-gnu.

Expected behavior with that version and triple appears to be an error message about the array constant being too large for the current architecture.  Specifically, per IRC:

```
error: the type `[u8; 18446744073709550615]` is too big for the current architecture 
```

Modifying the code in the very obvious manner to fix it makes it work fine.

**Comment by Aatch** [comment]
Duplicate of #30083. The "capacity overflow" is from trying to create the massive vector to pass to LLVM.
